Improve text accessibility

robcresswell opened the pull request:
This patch makes a few changes:

- Alter the "light" text colour so that it is WCAG AAA compliant for legibility
- Adjust code colours to be at least WCAG AA compliant.
- Darken the "medium" and "dark" colours, so as to preserve significance of
  different shades
- Make text use the font-size set by the browser, rather than hardcoding 15px.
  Most browsers default to 16px, in my understanding. Also adjust the max-width
  of the body, to account for the increased font-size and prevent aggressive
  line wrapping.
- Add VIM swap file extension to .gitignore.

robcresswell commented:
This is probably a bit of subjective / contentious change. It's also interesting that the Vue "green" completely fails the contrast test, but I didn't want to mess with Vue's visual identity. We could probably darken it (its usually on a bright white background) and tweak it a little without many people noticing, but that's really up to the cores.

chrisvfritz commented:
Thanks @robcresswell! This is something I'll want Evan to approve before merging, but before that we can get some of our designers to review. Also tagging @callumacrae since he specializes in accessibility.

phanan commented:
I've no opinion on the visuals yet, but adding VIM swap file extension to `.gitignore` isn't something I'd do. IMO such ignore rules should be set at the global, not project level.

phanan commented:
OK, I quite like the visual changes personally, and I think it does indeed improve readability. The foreground/background color contrast has been improved (from [9.28](http://contrast-ratio.com/#%2334495e-on-white) to [10.07](http://contrast-ratio.com/#%23304455-on-white)), and the new font size is more friendly I think. Over to @yyx990803, for branding/visual considerations.

robcresswell commented:
Just reverted the code font change that I realised I left in, as it was subjective, doesn't really affect accessibility, and alters the visual identity of the docs.

(Personally I would recommend using native font stacks for docs like these as they are technical in nature, not company-branded, and it reduces the network overhead in both bandwidth and latency, but thats a separate discussion)

callumacrae commented:
![image](https://user-images.githubusercontent.com/472830/40916545-0a5e74ba-67f8-11e8-92d0-315a59b9b595.png)

![image](https://user-images.githubusercontent.com/472830/40916554-0ea89078-67f8-11e8-9c44-ef8f4b694476.png)

Code blocks can now go off the right - I'm not sure if there was a convention keeping them on the page before though.

----

```html
<meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1, user-scalable=no">
```

Scrolling is disabled on mobile, would be great to deal with that: https://github.com/w3c/html/issues/602

Good work :)

robcresswell commented:
Thanks for the feedback @callumacrae; will take a look at both of those points over the next couple of days

sdras commented:
Hey @robcresswell! Any updates on this? Thanks for your help!

robcresswell commented:
Hey @sdras! Okay if I take a look over the weekend? To be honest, I'd forgotten I had this open, but I'm happy to work on it if its still deemed valuable? If not, feel free to close it.

sdras commented:
Yeah absolutely. Thanks @robcresswell!

robcresswell commented:
Okay, that might be the slowest update anyone has ever made to a PR, ever. Hopefully addressed some of the comments around line wrapping, and have also altered the meta to allow mobile zoom.

I'm really quite new to the a11y side of things, so please point me in the right direction if I've made some mistakes. @sdras, thanks for your patience after requesting an update. I should've asked for three weekends :innocent:

robcresswell commented:
For some side-by-side comparisons, to save reviewer time:
*BEFORE* https://vuejs.org/v2/guide/mixins.html#Custom-Option-Merge-Strategies
*AFTER* https://deploy-preview-1666--vuejs.netlify.com//v2/guide/mixins.html#Custom-Option-Merge-Strategies

*BEFORE* https://vuejs.org/v2/guide/transitions.html
*AFTER* https://deploy-preview-1666--vuejs.netlify.com/v2/guide/transitions.html

phanan commented:
I feel a bit uneasy with the increased font size for code which IMO is unbalanced with the paragraph text and looks crowded now with the 1.5em line-height. I've played along with the numbers and found this combination to be more balanced:

```css
code, pre {
  font-size: 0.85em;
}

pre code {
  line-height: 1.9;
  font-size: 1em;
}
```

Compare the screenshots of the current PR:

<img src=https://user-images.githubusercontent.com/8056274/57192381-13094a00-6f30-11e9-95c2-671d21f15018.png width="373"/>


And with my values, notice the inline `Vue.config.optionMergeStrategies`'s size and the line height of the code block:

<img src=https://user-images.githubusercontent.com/8056274/57192377-04229780-6f30-11e9-8cfb-69c3e0ca9527.png width="373"/>


WDYT?

robcresswell commented:
@phanan Agreed, that _is_ nicer. I'll update quickly.

robcresswell commented:
Okay:
- Slightly decreased the inline code size
- Increased the line height of the code blocks

https://github.com/vuejs/vuejs.org/compare/469b5658ae79d84d686c5383491ec98ee98cbb30..0bfc43c970f02771084c24897916a8ca1f66ad5b

robcresswell commented:
![image](https://user-images.githubusercontent.com/3816327/57194305-371f5800-6f3d-11e9-9659-b95d66267155.png)

robcresswell commented:
@sdras Oh no, I'm sorry :sweat_smile: Thanks for your patience :pray:
